move_base/include/move_base/move_base.h

```diff
@@ -76,6 +76,40 @@ namespace move_base {
     OSCILLATION_R
   };
 
+  //Used to calculate distance, time and average velocity
+  struct FeedbackInfo
+  {
+      FeedbackInfo()
+      {
+        clear();
+      }
+
+      void clear()
+      {
+        dist_to_goal = 0.0;
+        average_velocity = 0.0;
+        average_velocity_bias_correction = 0.0;
+        prev_time = ros::Time(0, 0);
+        average_velocity_iteration = 0;
+      }
+
+      double calculateTimeToGoal(double distance)
+      {
+        double epsilon = 0.01;
+        if (fabs(average_velocity_bias_correction - 0.0) < epsilon) {
+          return NAN;
+        }
+        else {
+          return fabs(distance/average_velocity_bias_correction);
```

Comment:
Again you take absolute, which gives time to goal for robot moving forward and backward the same. This should be:
```
if (average_velocity_bias_correction < epsilon)
    return NaN;
else
    return distance/average_velocity_bias_correction;
```

Reply:
In that way time to the will be negative